### PR TITLE
check if EditorAdaptersFactoryService gives us a null buffer

### PR DESCRIPTION
**Pull request by jmarolf**
fixes https://github.com/dotnet/roslyn/issues/43495

### CODE REVIEW

File: src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioActiveDocumentTracker.cs [hunk 1]
```diff
@@ -228,7 +228,7 @@ public FrameListener(VisualStudioActiveDocumentTracker service, IVsWindowFrame f
                     {
                         _textBuffer = _documentTracker._editorAdaptersFactoryService.GetDocumentBuffer(bufferAdapter);
 
-                        if (!_textBuffer.ContentType.IsOfType(ContentTypeNames.RoslynContentType))
+                        if (_textBuffer != null && !_textBuffer.ContentType.IsOfType(ContentTypeNames.RoslynContentType))
```
Comment (jasonmalinowski) [inline src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioActiveDocumentTracker.cs:231]: What was the bufferAdapter? It's not a regular editor buffer adapter but some sort of WinForms specific doc data?

Comment (jmarolf) [reply src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioActiveDocumentTracker.cs:231]: in my testing it was a `Microsoft.VisualStudio.Editor.Implementation.VsTextBufferAdapter` that we got from the forms designer.

image

Comment (jmarolf) [reply src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioActiveDocumentTracker.cs:231]: So I don't think we can filter out getting a null buffer from the adapter type

Comment (davidwengier) [reply src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioActiveDocumentTracker.cs:231]: Not sure if it helps, but I used a VSShellUtilities method for getting a text buffer from a frame in CPS: https://devdiv.visualstudio.com/DevDiv/_git/CPS?path=%2Fsrc%2FMicrosoft.VisualStudio.ProjectSystem.VS.Implementation%2FEditor%2FProjectFrameOpenCloseListener.cs&version=GBmaster&line=223&lineEnd=228&lineStartColumn=1&lineEndColumn=1&lineStyle=plain

**Pull request merged**